src/Offer/Events/AbstractLabelEvent.php

```diff
@@ -46,7 +46,13 @@ public static function deserialize(array $data): AbstractLabelEvent
 
         return new static(
             $data['item_id'],
-            new Label($data['label'], $data['visibility'])
+            new Label(self::trimLineEndings($data['label']), $data['visibility'])
         );
     }
+
+    private static function trimLineEndings(string $labelName): string
```

Comment:
Maybe `trim` is not the best name, because a `trim` in PHP code typically manipulates the start and/or end of a string.